bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/json/JsonAddonService.java

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.addon.marketplace.internal.json;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openhab.core.addon.Addon;
+import org.openhab.core.addon.AddonEventFactory;
+import org.openhab.core.addon.AddonService;
+import org.openhab.core.addon.AddonType;
+import org.openhab.core.addon.marketplace.MarketplaceAddonHandler;
+import org.openhab.core.addon.marketplace.MarketplaceHandlerException;
+import org.openhab.core.addon.marketplace.internal.json.model.AddonEntryDTO;
+import org.openhab.core.config.core.ConfigurableService;
+import org.openhab.core.events.Event;
+import org.openhab.core.events.EventPublisher;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * This class is a {@link AddonService} retrieving JSON marketplace information.
+ *
+ * @author Yannick Schaus - Initial contribution
+ * @author Jan N. Klug - Refactored for JSON marketplaces
+ */
+@Component(immediate = true, configurationPid = "org.openhab.core.marketplace.json", //
+        property = Constants.SERVICE_PID + "=org.openhab.core.marketplace.json")
+@ConfigurableService(category = "system", label = "JSON 3rd Party Addon Service", description_uri = JsonAddonService.CONFIG_URI)
```

Comment:
From an UX perspective - it's the 3rd configurable service related to add-ons that will appear under "Settings > System Services" with barely any parameter which is beginning to become a lot and quite messy. If anything we should really think about reducing this proliferation and merging related configurations (I've talked about this before).
- Add-on Management + Community Marketplace + JSON 3rd Party Addon Service _(also note the difference in spelling of Add-on/Addon!)_ => Add-on Management
- Regional Settings + Ephemeris => Regional Settings
- Audio + Voice => Audio & Voice
- Maybe also Sitemap + Charts => ?

Maybe have configurable services for the sole purposes of providing the configuration and have the actual services inject them?

Reply:
I thought about that, too. The problem is that we - at least I know of no way to do it - group in the UI. If we merge everything under e.g. `system:addon` (which in general would be preferred IMO), it's not easily visible to the user which parameter affects which service.

Reply:
Would it help to introduce another level in der config description URI? E.g. `system:addon:management`, `system:addon:marketplace` and `system:addon:json`? The UI could then group by the second level (if present). 

The good thing to this approach is that it is fully backward compatible since the service PID stays the same and therefore the configuration is also used after the config description URI changed.